packages/tracing/src/browser/metrics.ts

```diff
@@ -69,62 +69,73 @@ export class MetricsInstrumentation {
+
+          if (transaction.op === 'navigation' && timeOrigin + startTime < transaction.startTimestamp) {
+            return;
           }
-          case 'mark':
-          case 'paint':
-          case 'measure': {
-            const startTimestamp = addMeasureSpans(transaction, entry, startTime, duration, timeOrigin);
-            if (tracingInitMarkStartTime === undefined && entry.name === 'sentry-tracing-init') {
-              tracingInitMarkStartTime = startTimestamp;
-            }
-
-            // capture web vitals
-
-            const firstHidden = getVisibilityWatcher();
-            // Only report if the page wasn't hidden prior to the web vital.
-            const shouldRecord = entry.startTime < firstHidden.firstHiddenTime;
 
-            if (entry.name === 'first-paint' && shouldRecord) {
-              logger.log('[Measurements] Adding FP');
-              this._measurements['fp'] = { value: entry.startTime };
-              this._measurements['mark.fp'] = { value: startTimestamp };
+          switch (entry.entryType) {
+            case 'navigation': {
+              addNavigationSpans(transaction, entry, timeOrigin);
+              responseStartTimestamp = timeOrigin + msToSec(entry.responseStart as number);
+              requestStartTimestamp = timeOrigin + msToSec(entry.requestStart as number);
+              break;
             }
-
-            if (entry.name === 'first-contentful-paint' && shouldRecord) {
-              logger.log('[Measurements] Adding FCP');
-              this._measurements['fcp'] = { value: entry.startTime };
-              this._measurements['mark.fcp'] = { value: startTimestamp };
+            case 'mark':
+            case 'paint':
+            case 'measure': {
+              const startTimestamp = addMeasureSpans(transaction, entry, startTime, duration, timeOrigin);
+              if (tracingInitMarkStartTime === undefined && entry.name === 'sentry-tracing-init') {
+                tracingInitMarkStartTime = startTimestamp;
+              }
+
+              // capture web vitals
+
+              const firstHidden = getVisibilityWatcher();
+              // Only report if the page wasn't hidden prior to the web vital.
+              const shouldRecord = entry.startTime < firstHidden.firstHiddenTime;
+
+              if (entry.name === 'first-paint' && shouldRecord) {
+                logger.log('[Measurements] Adding FP');
+                this._measurements['fp'] = { value: entry.startTime };
+                this._measurements['mark.fp'] = { value: startTimestamp };
+              }
+
+              if (entry.name === 'first-contentful-paint' && shouldRecord) {
+                logger.log('[Measurements] Adding FCP');
+                this._measurements['fcp'] = { value: entry.startTime };
+                this._measurements['mark.fcp'] = { value: startTimestamp };
+              }
+
+              if (this._measurements['fcp']?.value && entry.startTime > this._measurements['fcp'].value) {
```

Comment:
@onurtemizkan Thanks for looking into this. This if block may be better placed outside this switch statement so we can capture TBT of other performance entries.

--------

For browsers that support the [Long Tasks API](https://developer.mozilla.org/en-US/docs/Web/API/Long_Tasks_API), would that be a better source of TBT?

For browsers that do no support the Long Tasks API, we could fallback to this implementation.

Reply:
It seems that perfume relies on the long task api https://github.com/Zizzamia/perfume.js/blob/401cb83d392240b0847588ca1899acc77465d144/src/paint.ts#L19 when calculating.

See lighthouse implementation here: https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/computed/metrics/total-blocking-time.js#L27

Reply:
@dashed do you think that we would also need TTI if we are adding total blocking time?

Reply:
I'm thinking perhaps instead of TBT / TTI we should just capture long tasks as spans where the long task api is available, since TBT and TTI will be somewhat unreliable in real world applications. Also, I don't know this for sure, but the TTI implementation will likely stop us capturing TBT in most situations since the transaction wraps up well before 5 second quiescence period.

Reply:
@AbhiPrasad @k-fish (replaying a conversation I had with @k-fish on Slack)

Given our pageload transaction model, we may not need to track TTI. I'll rationalize this.

Since TBT is defined to be a measure of blocking time in between FCP and TTI. We expect `FCP timestamp <= TTI timestamp`.

TTI is defined as:

> TTI considers a page "reliably interactive" if the main thread has been free of long tasks for at least five seconds. 
> From https://web.dev/tbt/#how-does-tbt-relate-to-tti

For most pageload transactions, we will flush them before TTI can be measured. Hence, we expect:

```
FCP timestamp <= pageload transaction end timestamp < TTI timestamp
```

For the pageload transactions we generate, I anticipate that we will only be able to measure TBT in between FCP and the moment the pageload transaction is flushed.